python/ray/workflow/step_executor.py

```diff
@@ -111,9 +113,25 @@ def execute_workflow(workflow: "Workflow") -> "WorkflowExecutionResult":
     # Stage 1: prepare inputs
     workflow_data = workflow.data
     inputs = workflow_data.inputs
+    # If the outer workflow step skips checkpointing, it would
+    # update the checkpoint context of all inner steps except
+    # the output step, marking them "detached" from the DAG.
```

Comment:
A little bit hard to understand what does it mean here.
Could you give more details like: what's the output step in the nested workflow and what does detached mean.